disnake/types/activity.py

```diff
@@ -27,6 +27,8 @@
 
 from typing import List, Literal, Optional, TypedDict
 
+from typing_extensions import NotRequired
```

Comment:
it may be cool to shelter this import behind a 3.11 check (even though its in types so it doesn't *really* matter.

Reply:
Fair point, on the other hand we're importing `typing_extensions.Self` (also a 3.11 feature) in lots of files without a version check so I'd kinda like to keep it this way.
We should properly add `typing_extensions` to our dependencies though - right now we're relying on the transitive dependency through aiohttp, which uses `typing_extensions >= 3.7.4`, but we're using features (not at runtime, tbf) that are only part of 4.x.